app/Services/LocationService.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+
+namespace App\Services;
+
+
+use App\Events\NewCityAdded;
+use App\Events\NewCountryAdded;
+use App\Events\NewStateAdded;
+use App\Models\Location\City;
+use App\Models\Location\Country;
+use App\Models\Location\State;
+
+class LocationService
+{
+    /**
+     * From an array of data, return $countryId
+     *
+     * @param array $addressArray
+     * @return Country
+     */
+    public function getCountryFromAddressArray(array $addressArray)
+    {
+        $countryCode = $addressArray["country_code"] ?? '';
+        $country = $addressArray["country"] ?? '';
+
+        if (!$countryCode) {
+            return Country::where('country', 'error_country')->first();
+        }
+
+        $country = Country::select('id', 'country', 'shortcode')
+            ->firstOrCreate([
+                'shortcode' => $countryCode,
+                'country' => $country
+            ]);
```

Comment:
We only want to check for countryCode here

Reply:
True, I forgot about that. In that case, we might want to change the method entirely and not use first or create. Because if we only use the country code, the new country will have an empty full name.